connect-inject/handler.go

```diff
@@ -119,6 +119,45 @@ type Handler struct {
 	// registrations. It will be overridden by a specific annotation.
 	DefaultProtocol string
 
+	// EnableNamespaces indicates that a user is running Consul Enterprise
+	// with version 1.7+ which is namespace aware. It enables Consul namespaces,
+	// with injection into either a single Consul namespace or mirrored from
+	// k8s namespaces.
+	EnableNamespaces bool
+
+	// AllowK8sNamespacesList is a list of k8s namespaces to explicitly allow for
+	// injection. It supports the special character `*` which indicates that
+	// all k8s namespaces are eligible unless explicitly denied. This filter
+	// is applied before checking pod annotations.
+	AllowK8sNamespacesSet mapset.Set
+
+	// DenyK8sNamespacesList is a list of k8s namespaces to explicitly deny
+	// injection and thus service registration with Consul. An empty list
+	// means that no namespaces are removed from consideration. This filter
+	// takes precedence over AllowK8sNamespacesList.
+	DenyK8sNamespacesSet mapset.Set
+
+	// ConsulNSName is the name of the Consul namespace to register all
+	// injected services into if Consul namespaces are enabled and mirroring
+	// is disabled. This will not be used if mirroring is enabled.
+	ConsulNSName string
```

Comment:
I'd use `Namespace` here instead of NS since you're using namespace elsewhere in these fields

Reply:
Updated.